src/components/v2/deployment-history/ea/EADeploymentHistory.component.tsx

```diff
@@ -320,27 +334,86 @@ function ExternalAppDeploymentHistory({ appId }: { appId: string }) {
                                         </Tippy>
                                     </div>
                                 )
-                            })
-                        }
+                            })}
+                        </div>
                     </div>
+                    {selectedDeploymentHistoryIndex !== 0 && (
+                        <Tippy className="default-tt" arrow={false} content={'Re-deploy this version'}>
+                            <button
+                                className="flex cta deploy-button"
+                                onClick={() => setShowRollbackConfirmation(true)}
+                            >
+                                <DeployButton className="deploy-button-icon" />
+                                <span className="ml-4">Deploy</span>
+                            </button>
+                        </Tippy>
+                    )}
                 </div>
             </div>
-        </div>
+        )
     }
 
+    async function handleDeployClick() {
+        try {
+            const selectedDeploymentHistory = deploymentHistoryArr[selectedDeploymentHistoryIndex]
+            const requestPayload: RollbackReleaseRequest = {
+                hAppId: appId,
+                version: selectedDeploymentHistory.version,
+            }
+
+            if (installedAppInfo) {
+                requestPayload.installedAppId = installedAppInfo.installedAppId
+                requestPayload.installedAppVersionId = installedAppInfo.installedAppVersionId
+            }
+
+
+            setShowRollbackConfirmation(false)
+            history.push(`${url.split('/').slice(0, -1).join('/')}/${URLS.APP_DETAILS}?refetchData=true`)
```

Comment:
Shouldn't we use history push after calling receiving success response of rollbackApplicationDeployment?

Reply:
Yes. was waiting for sequence confirmation. Now it'll be 

1. On confirmation, show loader on the button as waiting for state
2. On success, redirect the user & show toast
3. On failure, show error toast

Reply:
Done